sound/soc/intel/common/sof-function-topology-lib.c

```diff
@@ -55,9 +56,24 @@ int sof_sdw_get_tplg_files(struct snd_soc_card *card, const struct snd_soc_acpi_
 			tplg_dev = TPLG_DEVICE_SDCA_JACK;
 			tplg_dev_name = "sdca-jack";
 		} else if (strstr(dai_link->name, "SmartAmp")) {
-			tplg_dev = TPLG_DEVICE_SDCA_AMP;
-			tplg_dev_name = devm_kasprintf(card->dev, GFP_KERNEL,
-						       "sdca-%damp", dai_link->num_cpus);
+			/* If the AMP feedback topology is already used, continue */
+			if (tplg_mask & BIT(TPLG_DEVICE_SDCA_AMP_FEEDBACK))
+				continue;
+			if (strstr(dai_link->name, "Playback")) {
+				tplg_dev = TPLG_DEVICE_SDCA_AMP;
+				tplg_dev_name = devm_kasprintf(card->dev, GFP_KERNEL,
+							       "sdca-%damp", dai_link->num_cpus);
+			} else { /* Capture */
+				tplg_dev = TPLG_DEVICE_SDCA_AMP_FEEDBACK;
+				tplg_dev_name = devm_kasprintf(card->dev, GFP_KERNEL,
+							       "sdca-%damp-feedback",
+							       dai_link->num_cpus);
+				/* Remove the AMP topology, just use the Amp feedback topology */
+				if (tplg_mask & BIT(TPLG_DEVICE_SDCA_AMP)) {
+					tplg_mask &= ~BIT(TPLG_DEVICE_SDCA_AMP);
```

Comment:
@bardliao is this correct? Shouldn't we be replacing the entry in the tplg_files array with the amp feedback topology instead of simply updating the count and mask?